core/connection/manager.go

```diff
@@ -417,6 +417,22 @@ func (manager *connectionManager) onStateChanged(state State) {
 	}
 }
 
+func (manager *connectionManager) setupTrafficBlock(disableTrafficBlocking bool) error {
+	if disableTrafficBlocking {
+		return nil
+	}
+
+	removeRule, err := firewall.BlockNonTunnelTraffic(firewall.Session)
```

Comment:
Wanna do it as singleton, does it align with started `firewall.iptablesKillSwitch` struct?

Reply:
killSwitch* stuff completely removed